Dining/views.py

```diff
@@ -51,154 +84,84 @@ def process_date(context, day, month, year):
     return current_date
 
 
+# Todo: deprecate (possibly replace with a method on DiningListManager
 def get_list(current_date, identifier):
     """
-    Returns the dining list for the given data and identifier
-    :param current_date:
-    :param identifier:
-    :return:
+    Returns the dining list for the given data and identifier.
     """
-    # Get the dining list by the id of the association, shorthand form of the association or the person claimed
-    try:
-        if identifier is int:
-            return DiningList.objects.get(date=current_date, association_id=identifier)
-        else:
-            raise ObjectDoesNotExist("identifier is not an integer")
-    except ObjectDoesNotExist:
-        try:
-            return DiningList.objects.get(date=current_date, association__associationdetails__shorthand=identifier)
-        except ObjectDoesNotExist:
-            try:
-                return DiningList.objects.get(date=current_date, claimed_by__username=identifier)
-            except ObjectDoesNotExist:
-                # No proper identifier supplied
-                return None
+    return DiningList.objects.get(date=current_date, association__associationdetails__shorthand=identifier)
 
 
-class IndexView(View):
+class DayView(AbstractDayView):
```

Comment:
Ok. Ik ben voor het onderbrengen van abstract klassen waar mogelijk. (had het zelf in de diningslot ook al gedaan). Dat gezegd hebbende ben ik het niet eens met jouw aanpassingen van de volgende en vorige dag. De orginele implementatie zorgde ervoor dat er een bericht kwam als je in het weekend bezochte je niet slots aan kan maken etc, maar vanaf daar wel bladeren. Van wat ik zie in jouw nieuwe implementatie krijg je een 404, maar tevens ook geen mogelijkheid om naar andere dagen te gaan. (als ik de code goed lees).
Ook lijken de limieten op dag-browsen weg. Ik had deze ingevoegd zodat men niet perongeluk op een andere dinsdag inschrijft, maar wel de vrijheid heeft om een aantal dagen vooruit, dan wel achteruit te gaan. Voor specefieke data kan dit altijd in de url worden aangepast, maar dit is dan ook een bewuste keuze van de gebruiker. (overigens heb ik al een keer meegemaakt dan iemand bijna 2 weken van tevoren een slot had geclaimed om te koken, en hier een dag van tevoren aan herrinnerd moest worden waarna hij niet meer kon koken en de lijst werd geannuleerd.) Ik zou die limieten dan ook graag terug willen zien in implementatie.

Reply:
> Ok. Ik ben voor het onderbrengen van abstract klassen waar mogelijk. (had het zelf in de diningslot ook al gedaan). Dat gezegd hebbende ben ik het niet eens met jouw aanpassingen van de volgende en vorige dag. De orginele implementatie zorgde ervoor dat er een bericht kwam als je in het weekend bezochte je niet slots aan kan maken etc, maar vanaf daar wel bladeren. Van wat ik zie in jouw nieuwe implementatie krijg je een 404, maar tevens ook geen mogelijkheid om naar andere dagen te gaan. (als ik de code goed lees).

Klopt, je krijgt een 404 en dan kan je idd niet naar een andere dag bladeren. Omdat je d.m.v. browsen/linkjes nooit op een weekend dag komt wordt die pagina bij normaal gebruik dus nooit bereikt. Het aanpassen van de url in de browser zie ik niet als iets dat we moeten faciliteren, als dat nodig is voor iets (misschien om naar een specifiieke datum te gaan) denk ik dat het beter is om daar een user requirement voor te schrijven en daarna op een andere manier te implementeren, bijv. via een datepicker.

> Ook lijken de limieten op dag-browsen weg. Ik had deze ingevoegd zodat men niet perongeluk op een andere dinsdag inschrijft, maar wel de vrijheid heeft om een aantal dagen vooruit, dan wel achteruit te gaan. Voor specefieke data kan dit altijd in de url worden aangepast, maar dit is dan ook een bewuste keuze van de gebruiker. (overigens heb ik al een keer meegemaakt dan iemand bijna 2 weken van tevoren een slot had geclaimed om te koken, en hier een dag van tevoren aan herrinnerd moest worden waarna hij niet meer kon koken en de lijst werd geannuleerd.) Ik zou die limieten dan ook graag terug willen zien in implementatie.

Ja, lijkt mij ook nuttig, deze staat alleen nog niet bij de user requirements, ik zal deze er als suggestie bij zetten zodat we daarover kunnen discussieren.

Reply:
Als normale gebruikers in het weekend naar de website gaan maken ze gebruik van een weekend pagina. Daarom wil ik wel navigatie om van een weekend dag af te gaan en in plaats daarvan vooruit dan wel achteruit te gaan.

Reply:
Wat er nu gebeurt is als je in het weekend de website bezoekt ga je vanzelf naar de eerstvolgende maandag